runtime/compiler/runtime/SymbolValidationManager.cpp

```diff
@@ -0,0 +1,2329 @@
+
+   if (getIDFromSymbol(beholder) == 0)
+      {
+      TR_ASSERT(false, "beholder 0x%p should have already been validated\n", beholder);
+      TR::comp()->failCompilation<J9::AOTSymbolValidationManagerFailure>("Failed to validate in addClassFromITableIndexCPRecord");
+      }
+
+   int32_t arrayDims = 0;
+   clazz = getBaseComponentClass(clazz, arrayDims);
+
+   SymbolValidationRecord *record = new (_region) ClassFromITableIndexCPRecord(clazz, beholder, cpIndex);
+   return storeValidationRecordIfNecessary(static_cast<void *>(clazz), record, arrayDims);
+   }
+
+bool
+TR::SymbolValidationManager::addDeclaringClassFromFieldOrStaticRecord(TR_OpaqueClassBlock *clazz, J9ConstantPool *constantPoolOfBeholder, int32_t cpIndex)
+   {
+   if (!clazz)
+      return false;
+   if (inHeuristicRegion())
+      return true;
+
+   TR_OpaqueClassBlock *beholder = reinterpret_cast<TR_OpaqueClassBlock *>(J9_CLASS_FROM_CP(constantPoolOfBeholder));
+
+   if (getIDFromSymbol(beholder) == 0)
+      {
+      TR_ASSERT(false, "beholder 0x%p should have already been validated\n", beholder);
+      TR::comp()->failCompilation<J9::AOTSymbolValidationManagerFailure>("Failed to validate in addDeclaringClassFromFieldOrStaticRecord");
+      }
+
+   int32_t arrayDims = 0;
+   clazz = getBaseComponentClass(clazz, arrayDims);
+
+   SymbolValidationRecord *record = new (_region) DeclaringClassFromFieldOrStaticRecord(clazz, beholder, cpIndex);
+   return storeValidationRecordIfNecessary(static_cast<void *>(clazz), record, arrayDims);
+   }
+
+bool
+TR::SymbolValidationManager::addClassClassRecord(TR_OpaqueClassBlock *classClass, TR_OpaqueClassBlock *objectClass)
+   {
+   if (!classClass)
+      return false;
+   if (inHeuristicRegion())
+      return true;
+
+   if (getIDFromSymbol(objectClass) == 0)
+      {
+      TR_ASSERT(false, "objectClass 0x%p should have already been validated\n", objectClass);
+      TR::comp()->failCompilation<J9::AOTSymbolValidationManagerFailure>("Failed to validate in addClassClassRecord");
+      }
+
+   SymbolValidationRecord *record = new (_region) ClassClassRecord(classClass, objectClass);
+   return storeValidationRecordIfNecessary(static_cast<void *>(classClass), record);
+   }
+
+bool
+TR::SymbolValidationManager::addConcreteSubClassFromClassRecord(TR_OpaqueClassBlock *childClass, TR_OpaqueClassBlock *superClass)
+   {
+   if (!superClass)
+      return false;
+   if (inHeuristicRegion())
+      return true;
+
+   int32_t arrayDims = 0;
+   childClass = getBaseComponentClass(childClass, arrayDims);
+
+   if (getIDFromSymbol(superClass) == 0)
+      {
+      TR_ASSERT(false, "superClass 0x%p should have already been validated\n", superClass);
+      TR::comp()->failCompilation<J9::AOTSymbolValidationManagerFailure>("Failed to validate in addConcreteSubClassFromClassRecord");
+      }
+
+   SymbolValidationRecord *record = new (_region) SuperClassFromClassRecord(childClass, superClass);
```

Comment:
Should this be a `ConcreteSubClassFromClassRecord`?

Reply:
:scream: yes it should...

Reply:
For any reviewers reading this; the consequence of this error isn't anything functional except for that it would result in an AOT load failure.